webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/FileSystemCredentials.scala

```diff
@@ -1,23 +1,50 @@
 package com.scalableminds.webknossos.datastore.storage
 
-import play.api.libs.json.{Json, OFormat}
+import play.api.libs.json.{JsValue, Json, OFormat}
 
-sealed trait AnyCredential
+sealed trait FileSystemCredential
 
-object AnyCredential {
-  implicit val jsonFormat: OFormat[AnyCredential] = Json.format[AnyCredential]
+object FileSystemCredential {
+  implicit val jsonFormat: OFormat[FileSystemCredential] = Json.format[FileSystemCredential]
 }
 
 case class HttpBasicAuthCredential(name: String, username: String, password: String, user: String, organization: String)
-    extends AnyCredential
+    extends FileSystemCredential
 
 object HttpBasicAuthCredential {
   implicit val jsonFormat: OFormat[HttpBasicAuthCredential] = Json.format[HttpBasicAuthCredential]
 }
 
-case class S3AccessKeyCredential(name: String, keyId: String, key: String, user: String, organization: String)
-    extends AnyCredential
+case class S3AccessKeyCredential(name: String,
+                                 accessKeyId: String,
+                                 secretAccessKey: String,
+                                 user: String,
+                                 organization: String)
+    extends FileSystemCredential
 
 object S3AccessKeyCredential {
   implicit val jsonFormat: OFormat[S3AccessKeyCredential] = Json.format[S3AccessKeyCredential]
 }
+
+case class GoogleServiceAccountCredential(name: String, secretJson: JsValue, user: String, organization: String)
```

Comment:
I am not sure about the naming here. In the enum CredentialType there are HTTP_Basic_Auth and S3_Access_Key, which have the same name as the matching case classes (but with Credential) added here, while the Credential is called GoogleServiceAccountCredential and the type is called `CredentialType.GCS`. Maybe rename the enum type if this is the correct name?